### ISSUE TITLE
3.8-14.03 broke Samba

### ISSUE THREAD

**Issue by Mazo** [rockstor/rockstor-core#1385]
See thread here from another user: https://forum.rockstor.com/t/3-8-14-03-broke-samba/1726

            Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/rest_framework_custom/generic_view.py", line 40, in _handle_exception
    yield
  File "/opt/rockstor/src/rockstor/smart_manager/views/samba_service.py", line 44, in get
    config = get_global_config()
  File "/opt/rockstor/src/rockstor/system/samba.py", line 179, in get_global_config
    config[fields[0].strip()] = fields[1].strip()
IndexError: list index out of range

**Comment by phillxnet** [comment]
@Mazo Thanks for reporting this, I'm having a quick look now and think it's the indented comments within  the [global] section of /etc/samba/smb.conf thats causing a simple parsing error.
Currently working on a fix and should be available pretty soon.

**Comment by phillxnet** [comment]
With the above logging commit we get the following:
```
[09/Jul/2016 18:38:10] DEBUG [system.samba:182] FIELDS = ['workgroup ', ' mygroup']
[09/Jul/2016 18:38:10] DEBUG [system.samba:183] LEN OF FIELDS = 2
[09/Jul/2016 18:38:10] DEBUG [system.samba:182] FIELDS = ['server string ', ' Samba Server Version %v']
[09/Jul/2016 18:38:10] DEBUG [system.samba:183] LEN OF FIELDS = 2
[09/Jul/2016 18:38:10] DEBUG [system.samba:182] FIELDS = [';\tnetbios name ', ' MYSERVER']
[09/Jul/2016 18:38:10] DEBUG [system.samba:183] LEN OF FIELDS = 2
[09/Jul/2016 18:38:10] DEBUG [system.samba:182] FIELDS = [';\tinterfaces ', ' lo eth0 192.168.12.2/24 192.168.13.2/24']
[09/Jul/2016 18:38:10] DEBUG [system.samba:183] LEN OF FIELDS = 2
[09/Jul/2016 18:38:10] DEBUG [system.samba:182] FIELDS = [';\thosts allow ', ' 127. 192.168.12. 192.168.13.']
[09/Jul/2016 18:38:10] DEBUG [system.samba:183] LEN OF FIELDS = 2
[09/Jul/2016 18:38:10] DEBUG [system.samba:182] FIELDS = [';\tmax protocol ', ' SMB2']
[09/Jul/2016 18:38:10] DEBUG [system.samba:183] LEN OF FIELDS = 2
[09/Jul/2016 18:38:10] DEBUG [system.samba:182] FIELDS = ['# log files split per-machine:']
[09/Jul/2016 18:38:10] DEBUG [system.samba:183] LEN OF FIELDS = 1
[09/Jul/2016 18:38:10] ERROR [storageadmin.util:47] exception: list index out of range
Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/rest_framework_custom/generic_view.py", line 40, in _handle_exception
    yield
  File "/opt/rockstor/src/rockstor/smart_manager/views/samba_service.py", line 44, in get
    config = get_global_config()
  File "/opt/rockstor/src/rockstor/system/samba.py", line 186, in get_global_config
    config[fields[0].strip()] = fields[1].strip()
IndexError: list index out of range
[09/Jul/2016 18:38:10] DEBUG [storageadmin.util:48] Current Rockstor version: 3.8-14.03
```
Hence the failed parsing on indented remarks.

**Comment by phillxnet** [comment]
@schakrava With the 'final field count sanity check' patch:
https://github.com/phillxnet/rockstor-core/commit/4ddd2aae5f29e959e36542bebff878a5695ac238
the new code did retrieve [global] options and display them in the new custom options box and didn't trip up on the indented remarks but upon submitting samba config the existing /etc/samba/smb.conf file was found to be empty.

Handing over to you on this one as I suspect you know what's happened on this one already.

**Comment by phillxnet** [comment]
Extracted "Custom global configuration" displayed with testing channel released code once all indented remarks are removed ie lines such as:
```
        # log files split per-machine:
        # maximum size of 50KB per log file, then rotate:
```
Looks like this (note the indent and last line with spaces):
```
              ;	max protocol = SMB2
              ;	map hidden = no
              ;	add machine script = /usr/sbin/useradd -n -c "Workstation (%u)" -M -d /nohome -s /bin/false "%u"
              ;	printcap name = lpstat
              ;	logon path = 
              ;	domain logons = yes
              ;	map archive = no
              load printers = yes
              ;	security = user
              ;	domain master = yes
              ;	delete user script = /usr/sbin/userdel "%u"
              ;	add user script = /usr/sbin/useradd "%u" -n -g users
              ;	wins support = yes
              ;	map system = no
              ;	passdb backend = tdbsam
              ;	delete user from group script = /usr/sbin/userdel "%u" "%g"
              ;	local master = no
              passdb backend = tdbsam
              ;	delete group script = /usr/sbin/groupdel "%g"
              workgroup = MYGROUP
              cups options = raw
              ;	realm = MY_REALM
              ;	password server = <NT-Server-Name>
              max log size = 50
              log file = /var/log/samba/log.%m
              ;	wins proxy = yes
              ;	hosts allow = 127. 192.168.12. 192.168.13.
              ;	dns proxy = yes
              ;	wins server = w.x.y.z
              ;	os level = 33
              server string = Samba Server Version %v
              ;	add group script = /usr/sbin/groupadd "%g"
              ;	logon script = %u.bat
              ;	map read only = no
              ;	store dos attributes = yes
              ;	interfaces = lo eth0 192.168.12.2/24 192.168.13.2/24
              ;	printing = cups
              security = user
              ;	preferred master = yes
              ;	netbios name = MYSERVER
            
```

**Comment by phillxnet** [comment]
Submitting the above, which results from removing the indented remarks then results in an empty smb.conf. Was looking for a user applicable edit of config as a quick work around on released testing channel code but looks like there more to this one that I initially though.

**Comment by phillxnet** [comment]
@schakrava pr ready for testing and review.

Please update the following forum thread with significant development on this issue:
https://forum.rockstor.com/t/3-8-14-03-broke-samba/1726

**Comment by schakrava** [comment]
Thanks @Mazo and @phillxnet and my apologies for sloppy changes to begin with.

**Comment by schakrava** [comment]
Fixed by #1386

**Comment by phillxnet** [comment]
The indicated forum thread has now been updated.
